python/cuml/internals/base.pyx

```diff
@@ -329,12 +360,36 @@ class Base(TagsMixin,
 
         # If we are input, get the type from the input
         if output_type == 'input':
-            output_type = cuml.common.input_utils.determine_array_type(inp)
+            output_type = cuml.internals.input_utils.determine_array_type(inp)
+
+        return output_type
+
+    def _get_output_mem_type(self, inp):
+        """
+        Method to be called by predict/transform methods of inheriting classes.
+        Returns the appropriate memory type depending on the type of the input,
+        class output type and global output type.
+        """
+
+        # Default to the global type
+        output_type = cuml.global_settings.output_type
+        mem_type = cuml.global_settings.memory_type
+
+        # If it's None, default to our type
+        if mem_type in (None, MemoryType.mirror):
+            output_type = self.output_type
+
+        # If we are input, get the type from the input
+        if output_type == 'input':
+            output_type = cuml.internals.input_utils.determine_array_type(inp)
+            mem_type = cuml.internals.input_utils.determine_array_memtype(
+                inp
+            )
```

Comment:
It seems like this function is not actually returning the `mem_type`. Is it not used or tested?

Reply:
It appears to be? Is there something specific that makes you think it's not actually returning the `mem_type`?

Reply:
I was referring to the function implemented here, it returns `output_type` on line 389 so I don't see the point in determining the `mem_type` on line 385. 
https://github.com/rapidsai/cuml/blob/7ae2f92203ed48360b57de9fac9d733f234acbc2/python/cuml/internals/base.pyx#L389